ruby/ext/google/protobuf_c/repeated_field.c

```diff
@@ -615,7 +615,7 @@ VALUE RepeatedField_init(int argc, VALUE* argv, VALUE _self) {
 
   self->type_info = TypeInfo_FromClass(argc, argv, 0, &self->type_class, &ary);
   self->array = upb_Array_New(arena, self->type_info.type);
-  ObjectCache_Add(self->array, _self);
+  ObjectCache_GetSet(self->array, _self);
```

Comment:
should we assert `_self == result` from this GetSet?  alternately, should we assign `_self` to the result?

Reply:
I'm fine either way. The assumption here is that we allocated our own, so we can't possible have a duplicate, but I may be wrong.

Reply:
yeah, do you mind adding the assert?

Reply:
Done.